linux_os/guide/services/ssh/ssh_server/sshd_use_directory_configuration/rule.yml

```diff
@@ -0,0 +1,35 @@
+documentation_complete: true
+
+prodtype: fedora,rhel9
+
+title: 'Distribute the SSH Server configuration to multiple files in a config directory.'
+
+description: |-
+    Make sure to have the <tt>Include /etc/ssh/sshd_config.d/*</tt> line in the <tt>/etc/ssh/sshd_config</tt> file.
+    Ideally, don't have any active configuration directives in that file, and distribute the service configuration
+    to several files in the <tt>/etc/ssh/sshd_config.d</tt> directory.
+
+rationale: |-
+    This form of distributed configuration is considered as a good practice, and as other sshd rules assume that directives in files in the <tt>/etc/ssh/sshd_config.d</tt> config directory are effective, there has to be a rule that ensures this.
+    Aside from that, having multiple configuration files makes the SSH Server configuration changes easier to partition according to the reason that they were introduced, and therefore it should help to perform merges of hardening updates.
+
+severity: medium
+
+identifiers:
+    cce@rhel9: CCE-84156-9
+
+# The second part of the sentence explaining what got wrong.
+# ... Is it the case that <YOUR ocil_clause>
+ocil_clause: "you don't include other configuration files from the main configuration file"
+
+# A setp-by-step guide how to modify the configuration to achieve compliance
+ocil: |-
+    To determine whether the SSH server includes configuration files from the right directory, run the following command:
+    <pre>$ sudo grep -i '^Include' /etc/ssh/sshd_config</pre>
+    If a line <tt>Include /etc/ssh/sshd_config.d/*</tt> is returned, then the configuration file inclusion is set correctly.
+
+template:
+    name: lineinfile
```

Comment:
I don't think this does what you want sadly :/

The reason the `sshd`-specific template was introduced was due to `Match` blocks. Reading the `sshd_config` man page:

> An `Include` directive may appear inside a `Match` block to perform conditional inclusion.

So you're almost better off leaving `sshd_lineinfile` alone, updating it to take an optional file and using `sshd_lineinfile` here and then stub out a `sshd_config` that does what you've updated below (either using `sshd_lineinfile` directly on `/etc/ssh/sshd_config` or on a config directory). :/

Reply:
I see. An alternative would be to extend this rule with a check that would make sure that there is no `Match` in the main config file - the direction set by default sshd config file in Fedora indicates that the preferable configuration should be done by the config files from the config directory, and that the main config file should just include them and do nothing else.
As an alternative to what you have suggested could be rebranding `sshd_lineinfile` to `sshd_config`, and just using macros for this rule. In other words, the `sshd_lineinfile` template in its current form would simply disappear - why to have a template, if there is only one rule using it?

Reply:
I don't know about other products (such as RHEL8 and the SUSE versions) but at least we're currently preferring the main config file in our CIS and STIG for Ubuntu 20.04... So to technically be STIG compliant here, we'd prefer to keep a `sshd_lineinfile` around that can set stuff (and is cognizant of `Match`) in the main `/etc/ssh/sshd_config` file... Even if we do move to `sshd_config.d` for other newer content eventually. 16.04 is just nearing the ESM phase (where we mostly intend to stop rebasing to new CIS & STIG versions and definitely won't be seeing a CaC backport) --- so I'd expect 20.04 to be around for the next 4/5 years too with the current ruleset. :/

Reply:
The change that I have introduced allows products to opt-in for the sshd config dir behavior. IOW, merging the PR shouldn't affect other products than Fedora and RHEL9, so a rule can behave differently based on the target product without changing its definition.

Reply:
Ah right, I saw that and ignored it. :facepalm: 

Yes, so then I think it is just a question of how you want to handle this one-off weird thingy. I think directly calling macros from `oval/...` and `bash/...` is as good as anything to me.

Reply:
(Or maybe introduce another rule as well, forbidding `Match` sections when this rule is selected?)

Reply:
Could you please point me out to handling of the `Match` block? I took a look into definitions in `macros-oval.jinja`, and I couldn't find anything like that there.

Reply:
For Ansible it was here:

https://github.com/ComplianceAsCode/content/blob/master/shared/macros-ansible.jinja#L145-L156

I don't know if it ever got finished for shell. I don't think it was considered for OVAL because regardless of whether or not a value is in a Match block or not, it should be a finding if the value is wrong. But if the value is correct and the option _has_ to be present, then it must live outside a Match block, which makes the logic a little harder.

At the time I was contributing, I was most familiar with Ansible/Bash and not familiar at all with OVAL.

So perhaps my comments here are only relevant to the Ansible portion of the template. 

I'm not quite following this PR right now, but I think you didn't touch the Ansible? At least, that's what its looking like to me -- in which case, the Ansible wouldn't support `sshd_config.d` folders.

Reply:
You are right that the new system won't work with Ansible. I have checked out what Ansible can do for us here, and I think that not much - there is still no sshd module, and the lineinfile module needs a filename rather than a regex with an indication of where to place potential lines. IOW, it looks like that we will need an Ansible "program" with variables, ifs (whens) and so on in order to handle this.
I wouldn't consider this lack of Ansible support as a PR blocker, because it is a missing feature rather than a introduced regression - a new configuration practice has appeared out there, and the content project just tries to catch up step by step.